R/ica.R

```diff
@@ -188,21 +188,23 @@ prep.step_ica <- function(x, training, info = NULL, ...) {
 bake.step_ica <- function(object, new_data, ...) {
   uses_dim_red(object)
 
-  if (object$num_comp > 0 && length(object$columns) > 0) {
-    check_new_data(object$columns, object, new_data)
+  if (object$num_comp == 0 || length(object$columns) == 0) {
```

Comment:
I feel like it would be same to assume that `length()` returns a integer. but you are correct with regards to `num_comp` which right now offers no checking